android/src/main/kotlin/com/mr/flutter/plugin/filepicker/FileUtils.kt

```diff
@@ -0,0 +1,615 @@
+        try {
+            context.contentResolver.openInputStream(originalImageUri).use { imageStream ->
+                val compressedFile = createImageFile(context, originalImageUri)
+                val originalBitmap = BitmapFactory.decodeStream(imageStream)
+                // Compress and save the image
+                val fos = FileOutputStream(compressedFile)
+                originalBitmap.compress(
+                    getCompressFormat(context, originalImageUri),
+                    compressionQuality,
+                    fos
+                )
+                fos.flush()
+                fos.close()
+                compressedUri = Uri.fromFile(compressedFile)
+            }
+        } catch (e: IOException) {
+            throw RuntimeException(e)
+        }
+        return compressedUri
+    }
+
+    @Throws(IOException::class)
+    private fun createImageFile(context: Context, uri: Uri): File {
+        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss", Locale.getDefault()).format(Date())
+        val imageFileName = "IMAGE_" + timeStamp + "_"
+        val storageDir = context.cacheDir
+        return File.createTempFile(imageFileName, "." + getFileExtension(context, uri), storageDir)
+    }
+
+    /**
+     * @param uri The Uri to check.
+     * @return Whether the Uri authority is DownloadsProvider.
+     */
+    fun isDownloadsDocument(uri: Uri): Boolean {
+        return "com.android.providers.downloads.documents" == uri.authority
+    }
+
+    @JvmStatic
+    fun clearCache(context: Context): Boolean {
+        try {
+            val cacheDir = File(context.cacheDir.toString() + "/file_picker/")
+            recursiveDeleteFile(cacheDir)
+        } catch (ex: Exception) {
+            Log.e(
+                TAG,
+                "There was an error while clearing cached files: $ex"
+            )
+            return false
+        }
+        return true
+    }
+
+    fun loadData(file: File, fileInfo: FileInfo.Builder) {
+        try {
+            val size = file.length().toInt()
+            val bytes = ByteArray(size)
+
+            try {
+                val buf = BufferedInputStream(FileInputStream(file))
+                buf.read(bytes, 0, bytes.size)
+                buf.close()
+            } catch (e: FileNotFoundException) {
+                Log.e(TAG, "File not found: " + e.message, null)
+            } catch (e: IOException) {
+                Log.e(TAG, "Failed to close file streams: " + e.message, null)
+            }
+            fileInfo.withData(bytes)
+        } catch (e: Exception) {
+            Log.e(
+                TAG,
+                "Failed to load bytes into memory with error $e. Probably the file is too big to fit device memory. Bytes won't be added to the file this time."
+            )
+        }
+    }
+
+    @JvmStatic
+    fun openFileStream(context: Context, uri: Uri, withData: Boolean): FileInfo? {
+        Log.i(TAG, "Caching from URI: $uri")
+        var `in`: InputStream? = null
```

Comment:
Can we rename these to `fileInputStream` and `fileOutputStream` ? "fos" sounds very vague anyway.